File: src/Factory/ShapeFactory.java
package Factory;

import Models.Circle;
import Models.Rectangle;
import Models.Shape;
import Models.Square;

public class ShapeFactory
{
  public Shape getShape(String shapeType)
  {
    if(shapeType.trim().length() != 0 && !shapeType.isEmpty())
    {
      if(shapeType.equalsIgnoreCase("Circle"))
      {
        return new Circle();
      }
      else if(shapeType.equalsIgnoreCase("Square"))
      {
        return new Square();
      }
      else if(shapeType.equalsIgnoreCase("Rectangle"))
      {
        return new Rectangle();
      }
    }
    else
    {
      return null;
    }
    return null;
  }
}

File: src/FactoryKt/ShapeFactory.kt
package FactoryKt

import KotlinModels.Circle
import KotlinModels.Rectangle
import KotlinModels.Shape
import KotlinModels.Square

class ShapeFactory
{
  fun getShape(shapeType: String): Shape?
  {
    if(!shapeType.trim().isEmpty() && shapeType != null)
    {
      if(shapeType.equals("Circle",true))
      {
        return Circle()
      }
      else if(shapeType.equals("Square",true))
      {
        return Square()
      }
      else if(shapeType.equals("Rectangle",true))
      {
        return Rectangle()
      }
    }
    else
    {
      return null
    }
    return null
  }

}
